processing_scripts/Workflow7_ImageJ_Jython_Analyst.py

```diff
@@ -0,0 +1,236 @@
+def openOmeroImage(ctx, image_id):
+    browse = gateway.getFacility(BrowseFacility)
+    image = browse.getImage(ctx, long(image_id))
+    pixels = image.getDefaultPixels()
+    sizeZ = pixels.getSizeZ()
+    sizeT = pixels.getSizeT()
+    sizeC = pixels.getSizeC()
+    sizeX = pixels.getSizeX()
+    sizeY = pixels.getSizeY()
+    pixtype = pixels.getPixelType()
+    pixType = FormatTools.pixelTypeFromString(pixtype)
+    bpp = FormatTools.getBytesPerPixel(pixType)
+    isSigned = FormatTools.isSigned(pixType)
+    isFloat = FormatTools.isFloatingPoint(pixType)
+    isLittle = False
+    interleave = False
+    
+    store = gateway.getPixelsStore(ctx)
+    pixelsId = pixels.getId()
+    store.setPixelsId(pixelsId, False)
+    stack = ImageStack(sizeX, sizeY)
+    for t in range(0,sizeT):
+        for z in range(0,sizeZ):
+            for c in range(0, sizeC):
+                plane = store.getPlane(z, c, t)
+
+                channel = ImageTools.splitChannels(plane, 0, 1, bpp, False, interleave)
+                pixels = DataTools.makeDataArray(plane, bpp, isFloat, isLittle)
+
+                q = pixels
+                if (len(plane) != sizeX*sizeY):
+                    tmp = q
+                    q = zeros(sizeX*sizeY, 'h')
+                    System.arraycopy(tmp, 0, q, 0, Math.min(len(q), len(tmp)))
+                    if isSigned:
+                        q = DataTools.makeSigned(q)
+                    
+                if q.typecode == 'b':
+                    ip = ByteProcessor(sizeX, sizeY, q, None)
+                elif q.typecode == 'h':
+                    ip = ShortProcessor(sizeX, sizeY, q, None)
+                stack.addSlice('', ip)
+    # Do something
+    imp = ImagePlus(image.getName(), stack)
+    imp.setDimensions(sizeC, sizeZ, sizeT)
+    imp.setOpenAsHyperStack(True)
+    imp.show()
+    return imp
+
+# upload image to OMERO server
+def upload_image(path, gateway, id):
+    "Upload an image to omero"
+
+    user = gateway.getLoggedInUser()
+    sessionKey = gateway.getSessionId(user)
+
+    config = ImportConfig()
+    config.debug.set('false')
+    config.hostname.set(HOST)
+    config.sessionKey.set(sessionKey)
+    value = "omero.model.Dataset:"
+    value += str(id)
+    config.target.set(value)
+
+    loci.common.DebugTools.enableLogging("DEBUG")
+
+    store = config.createStore()
+    reader = OMEROWrapper(config)
+
+    library = ImportLibrary(store, reader)
+    error_handler = ErrorHandler(config)
+
+    library.addObserver(LoggingImportMonitor())
+    candidates = ImportCandidates(reader, path, error_handler)
+    reader.setMetadataOptions(DefaultMetadataOptions(MetadataLevel.ALL))
+    return library.importCandidates(config, candidates)
+
+
+# Prototype analysis example
+gateway = connect_to_omero()
+ctx = SecurityContext(group_id)
+exp = gateway.getLoggedInUser()
+exp_id = exp.getId()
+
+# get all images_ids in an omero dataset
+dataset_id = 1107
```

Comment:
aha, so I should be encouraged to edit the Dataset ID too ? The whole script here seems to alternate between ``I work on one image`` and ``I work on whole dataset`` with no clearly defined boundaries between those two concepts ?